### DISCUSSION TITLE
Day of Year mode and Milliseconds

### DISCUSSION THREAD

**Discussion by umer936** [easepick/easepick#182] [Ideas]
Currently we use https://trentrichardson.com/examples/timepicker/#tp-examples . However, it is unmaintained and relies on jquery and jquery-ui, both of which I would like to drop.

In searching, I have been unable to find datetimepickers that support doy and/or millisecond support.

1. Day of Year
    trent richardson's timepicker supports doy through the flag 'DDD'. It parses the timestamp in and out of DOY when clicking the day in the selector. I'm not sure how much work that would require to add to https://easepick.com/packages/datetime.html#tokens-format. Optimal behavior would be an option that allows the selector to switch to a DOY view mode showing the doy number in the day boxes, but having it like this image is sufficient. 
![image](https://user-images.githubusercontent.com/2078163/232581217-b7f30673-8f5b-4136-8e83-02462400ab75.png)

2. Millisecond support
    our various sensors can have pretty fine-grained data fidelity, so millisecond support is necessary for us (smaller would be even better). This same issue is holding back phpmyadmin https://github.com/phpmyadmin/phpmyadmin/issues/17604 as well. This can be defined by the format 'HH:mm:ss.S+' from the unicode standard.

3. Sliders
    This doesn't seem to be the intended direction for this project, but I did like the sliders in the trentrichardson timepicker I linked above.

4. Typing
    I just noticed that it does not allow typing in the input field it is attached to. I would like to retain that option or maybe have the easepicker show if a button is clicked next to the input instead. I know I can just have another set of inputs and use an event to copy it over but I'm not seeing if this is already developed in here.

Thank you for the work on the project! 😃 It looks really nice. 
I'd love to be able to use it. Based on what y'all think is best, I can definitely work on these. I notice that there is plugin development. Would any of these better serve as plugins or as core? 

I've also opened a similar issue here: https://github.com/Eonasdan/tempus-dominus/issues/2588 and here: https://github.com/t1m0n/air-datepicker/issues/540

**Comment by trymeouteh** [comment]
Would like to see milliseconds as well.

**Reply by umer936** [reply]
@trymeouteh I ended up just making my own timepicker here: https://github.com/umer936/datetimepicker.js 

Feel free to try it out and give any feedback. I haven't had a chance to try it in a lot of different contexts so any feedback you have would be appreciated.
